src/readii_2_roqc/feature_extraction/extract.py

```
@@ -269,86 +269,95 @@ def compile_dataset_features(dataset_index: pd.DataFrame,
+
     if existing_dataset_files and not overwrite:
         # Get the image classes in the same format as readii_image_classes (a set of tuples)
         compiled_image_classes = {tuple(file.name.removesuffix('_features.csv').split('_')) for file in existing_dataset_files}
-        
+
+        # Load in the existing compiled dataset files into a dictionary to match function output
+        for file in existing_dataset_files:
+            image_class = file.name.removesuffix('_features.csv')
+            try:
+                compiled_dataset_features[image_class] = pd.read_csv(file)
+            except pd.errors.EmptyDataError as e:        
```

Ruff (0.13.3)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)